StripeCore/StripeCore/Source/Analytics/STPAnalyticEvent.swift

```diff
@@ -217,6 +217,7 @@ import Foundation
     case paymentSheetFormCompleted = "mc_form_completed"
     case paymentSheetCardNumberCompleted = "mc_card_number_completed"
     case paymentSheetDeferredIntentPaymentMethodIdMismatch = "mc_deferred_intent_payment_method_id_mismatch"
+    case paymentSheetDeferredIntentPaymentMethodFingerprintMismatch = "mc_deferred_intent_payment_method_fingerprint_mismatch"
```

Comment:
Let's just combine these into a single `paymentSheetDeferredIntentPaymentMethodMismatch` event and send more details in the analytic payload to distinguish between the specific failures (id mismatch vs fingerprint mismatch) by supplying `additionalNonPIIParams` in the `ErrorAnalytic` initializer. 

Whenever we change analytics we need to make sure we update relevant queries / make sure we don't break existing queries but in this case I don't think we specifically alert/search for this today.